### DISCUSSION TITLE
ArrayBatchformatter not compatible with Logstash/Elastic

### DISCUSSION THREAD

**Discussion by aboeHarry** [FantasticFiasco/serilog-sinks-http#264] [Ideas]
the DefaultBatchFormatter had the [JsonProperty("events")] the ArrayBatchFormatter doesn't.

It would be nice to have an EventsBatchFormatter (the former DefaultBatchFormatter). For those who need Logstash compatibility. Instead of a custom IBatchFormatter implementation.

**Comment by FantasticFiasco** [comment]
Hi @aboeHarry. I don't think having the `events` property is a requirement of Logstash. On the contrary I think the array batch formatter is easier to work with when it comes to the Elastic Stack. Please see the [Serilog and the Elastic Stack](https://github.com/FantasticFiasco/serilog-sinks-http-sample-elastic-stack) for an example where we use the sink with the default array batch formatter when sending events using Serilog.

Please let me know if I misunderstood your question.

**Reply by aboeHarry** [reply]
Hi @FantasticFiasco, thanks for your time! 
With the versions we have at our location the missing events property is causing the logging not be processed. This is managed by an other team so we will discuss this with them.

Thanks again!

Ali

**Reply by FantasticFiasco** [reply]
For the implementation of the `DefaultBatchFormatter`, please see the [wiki](https://github.com/FantasticFiasco/serilog-sinks-http/wiki/Batch-formatters#custom-formatter-2---defaultbatchformatter).
